lang/typescript/src/extended-address/splitAddress1WithFallback.test.ts

```diff
@@ -0,0 +1,80 @@
+import {splitAddress1WithFallback} from './splitAddress1WithFallback';
+
+describe('splitAddress1WithFallback', () => {
+  test('returns null when extended address is not defined for region', () => {
+    expect(splitAddress1WithFallback('US', '123 Main')).toBeNull();
+  });
+
+  test('returns empty object when extended address string is empty', () => {
+    expect(splitAddress1WithFallback('NL', '')).toEqual({});
+  });
+
+  test('returns street number if string before delimiter is empty', () => {
+    expect(splitAddress1WithFallback('NL', '\u2060123')).toEqual({
+      streetNumber: '123',
+    });
+  });
+
+  test('returns full address object when separated by delimiter', () => {
+    expect(splitAddress1WithFallback('NL', 'Main \u2060123')).toEqual({
+      streetName: 'Main',
+      streetNumber: '123',
+    });
+  });
+
+  test('returns address1 as street name when not separated by delimiter and no address1 regex defined', () => {
+    expect(splitAddress1WithFallback('MX', 'Main 123')).toEqual({
+      streetName: 'Main 123',
+    });
+  });
+
+  test('returns full address object when not separated by delimiter but matches address1 regex', () => {
+    const testCases = [
+      {
+        input: {country: 'NL', address: 'Mercuriusstraat 26'},
+        expected: {streetName: 'Mercuriusstraat', streetNumber: '26'},
+      },
+      {
+        input: {country: 'NL', address: 'Bloemgracht 41B'},
+        expected: {streetName: 'Bloemgracht', streetNumber: '41B'},
+      },
+      {
+        input: {country: 'NL', address: 'Bloemgracht 41b'},
+        expected: {streetName: 'Bloemgracht', streetNumber: '41b'},
+      },
+      {
+        input: {country: 'NL', address: 'Meester Arendstraat 48 B'},
+        expected: {streetName: 'Meester Arendstraat', streetNumber: '48 B'},
+      },
+    ];
+
+    testCases.forEach(({input, expected}) => {
+      expect(splitAddress1WithFallback(input.country, input.address)).toEqual(
+        expected,
+      );
+    });
+  });
+
+  test('returns address1 as street name when not separated by delimiter and does not match address1 regex', () => {
+    const testCases = [
+      {
+        input: {country: 'NL', address: 'Kempenaar 25 11'},
+        expected: {streetName: 'Kempenaar 25 11'},
+      },
+      {
+        input: {country: 'NL', address: '40 Baandersstraat'},
+        expected: {streetName: '40 Baandersstraat'},
+      },
+      {
+        input: {country: 'BR', address: 'Main, 123, Apt 2'},
+        expected: {streetName: 'Main, 123, Apt 2'},
+      },
+    ];
+
+    testCases.forEach(({input, expected}) => {
+      expect(splitAddress1WithFallback(input.country, input.address)).toEqual(
+        expected,
+      );
+    });
+  });
+});
```

Comment:
For this and the above test that handles multiple test cases Vitest has a utility called [`test.each`](https://vitest.dev/api/#test-each). You can pass the inputs as an object to `test.each` and provide a clean way to print the results for each test case. I'm not sure if you can pass objects as input though so you might have to do flatten the input objects:

```ts
// Nested objects might work, but I'm not 100% sure
test.each([
  { input: { country: 'NL', address: 'Kempenaar 25 11' }, expected: 'Kempenaar 25 11' }
])('splitAddress1WithFallback($input.country, $input.country) -> $expected', ({ input, expected }) => {
  expect(splitAddress1WithFallback(input.country, input.address)).toBe(expected)
})

// Flattened, definitely will work
test.each([
  { country: 'NL', address: 'Kempenaar 25 11', expected: 'Kempenaar 25 11' }
])('splitAddress1WithFallback($country, $address) -> $expected', ({ country, address, expected }) => {
  expect(splitAddress1WithFallback(country, address)).toBe(expected)
})
```

Reply:
Oh nice! Applied 👍